gammapy/maps/wcs/core.py

```diff
@@ -302,3 +305,87 @@ def _make_hdu_sparse(data, npix, hdu, header):
             ]
 
         return fits.BinTableHDU.from_columns(cols, header=header, name=hdu)
+
+    def containment_region(self, fraction=0.68, apply_union=True):
+        """Find the iso-contours region corresponding to a given containment
+            for a map of integral quantities.
+
+        Parameters
+        ----------
+        self : `~gammapy.maps.WcsNDMap`
+            Map of integral quantities
+        fraction : float
+            Containment fraction
+        apply_union : bool
+            It True return a compound region otherwise return a list of polygon regions.
+            Default is True.
+
+        Returns
+        -------
+        regions : list of ~regions.PolygonSkyRegion` or `~regions.CompoundSkyRegion`
+            regions from iso-contours matching containment fraction
+        """
+        fmax = np.nanmax(self.data)
+        if fmax > 0.0:
+            ordered = np.sort(self.data.flatten())[::-1]
+            cumsum = np.nancumsum(ordered)
+            ind = np.argmin(np.abs(cumsum / cumsum.max() - fraction))
+            fval = ordered[ind]
+
+            plt.ioff()
+            fig = plt.figure()
+            cs = plt.contour(self.data.squeeze(), [fval])
+            plt.close(fig)
+            plt.ion()
+            regions_pieces = []
+            for kp, pp in enumerate(cs.collections[0].get_paths()):
+                vertices = []
+                for v in pp.vertices:
+                    v_coord = self.geom.pix_to_coord(v)
+                    vertices.append([v_coord[0], v_coord[1]])
+                vertices = SkyCoord(vertices, frame=self.geom.frame)
+                regions_pieces.append(PolygonSkyRegion(vertices))
+
+            if apply_union:
+                # compound from union seems not supported to write in ds9 format
```

Comment:
Should we add this ds9 issue into the docstring?

Reply:
done